mlir/cmake/modules/AddMLIR.cmake

```diff
@@ -584,7 +584,7 @@ function(add_mlir_aggregate name)
   # TODO: Should be transitive.
   set_target_properties(${name} PROPERTIES
     MLIR_AGGREGATE_EXCLUDE_LIBS "${_embed_libs}")
-  if(MSVC)
+  if(WIN32)
```

Comment:
this one and below isn't strictly related to the build failure but it's also wrong - this is a windows platform requirement, not an `MSVC` requirement.